pkg/api/db.go

```diff
@@ -73,6 +73,17 @@ func CreateVolumeDBEntry(ctx *c.Context, in *model.VolumeSpec) (*model.VolumeSpe
 	return db.C.CreateVolume(ctx, in)
 }
 
+func CreateVolumeError(ctx *c.Context, in *model.VolumeSpec) {
+	var errMsg = "Not able to connect to controller"
+	log.Error(errMsg)
+	in.UserId = ctx.UserId
+	in.Status = model.VolumeError
```

Comment:
IMHO, it is better we define another status here called ```errorCreating```. This is the actual error right, some error during creating of the volume, which is different from the ```volumeError``` that we already have.